### ISSUE TITLE
Harmony & Roxas commits not live?

### ISSUE THREAD

**Issue by Ayukito** [rileytestut/Delta#9]
https://github.com/rileytestut/Delta/tree/master/External
Going to these commits error out, and trying to clone the repo submodules fails as a result

edit:

> error: Server does not allow request for unadvertised object 4d7ba40073c8fed53fa92f7db07d75b4b8929a41
Fetched in submodule path 'External/Harmony', but it did not contain 4d7ba40073c8fed53fa92f7db07d75b4b8929a41. Direct fetching of that commit failed.

**Comment by GoldenJoe** [comment]
Here's my error log:

```
git -c filter.lfs.smudge= -c filter.lfs.required=false -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree clone https://github.com/rileytestut/Delta /Users/GoldenJoe/Development/Xcode/Delta 
Cloning into '/Users/GoldenJoe/Development/Xcode/Delta'...
warning: templates not found /usr/local/git/share/git-core/templates


git -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree submodule update --init --recursive 
Cloning into 'Cores/DSDeltaCore'...
warning: templates not found /usr/local/git/share/git-core/templates
Cloning into 'desmume'...
warning: templates not found /usr/local/git/share/git-core/templates
Cloning into 'Cores/DeltaCore'...
warning: templates not found /usr/local/git/share/git-core/templates
Warning: Permanently added the RSA host key for IP address '140.82.114.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
Submodule 'Cores/DSDeltaCore' (https://github.com/rileytestut/DSDeltaCore.git) registered for path 'Cores/DSDeltaCore'
Submodule 'Cores/DeltaCore' (git@github.com:rileytestut/DeltaCore.git) registered for path 'Cores/DeltaCore'
Submodule 'Cores/GBADeltaCore' (git@github.com:rileytestut/GBADeltaCore.git) registered for path 'Cores/GBADeltaCore'
Submodule 'Cores/GBCDeltaCore' (git@github.com:rileytestut/GBCDeltaCore.git) registered for path 'Cores/GBCDeltaCore'
Submodule 'Cores/N64DeltaCore' (git@github.com:rileytestut/N64DeltaCore.git) registered for path 'Cores/N64DeltaCore'
Submodule 'Cores/NESDeltaCore' (git@github.com:rileytestut/NESDeltaCore.git) registered for path 'Cores/NESDeltaCore'
Submodule 'Cores/SNESDeltaCore' (git@github.com:rileytestut/SNESDeltaCore.git) registered for path 'Cores/SNESDeltaCore'
Submodule 'External/Harmony' (https://github.com/rileytestut/Harmony.git) registered for path 'External/Harmony'
Submodule 'External/Roxas' (git@github.com:rileytestut/Roxas.git) registered for path 'External/Roxas'
Submodule path 'Cores/DSDeltaCore': checked out '40d515a2088be952b6c4038ff715d1b91ce0667f'
Submodule 'desmume' (https://github.com/TASVideos/desmume.git) registered for path 'desmume'
Submodule path 'Cores/DSDeltaCore/desmume': checked out '9e6b4f128f45f84155c6e971e464de84c88986a7'
Cloning into 'Cores/DeltaCore'...
warning: templates not found /usr/local/git/share/git-core/templates
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
Cloning into 'Cores/DeltaCore'...
warning: templates not found /usr/local/git/share/git-core/templates
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
Cloning into 'Cores/DeltaCore'...
warning: templates not found /usr/local/git/share/git-core/templates
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
Completed with errors, see above
```

**Comment by Ayukito** [comment]
> Here's my error log:
> 
> ```
> git -c filter.lfs.smudge= -c filter.lfs.required=false -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree clone https://github.com/rileytestut/Delta /Users/GoldenJoe/Development/Xcode/Delta 
> Cloning into '/Users/GoldenJoe/Development/Xcode/Delta'...
> warning: templates not found /usr/local/git/share/git-core/templates
> 
> 
> git -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree submodule update --init --recursive 
> Cloning into 'Cores/DSDeltaCore'...
> warning: templates not found /usr/local/git/share/git-core/templates
> Cloning into 'desmume'...
> warning: templates not found /usr/local/git/share/git-core/templates
> Cloning into 'Cores/DeltaCore'...
> warning: templates not found /usr/local/git/share/git-core/templates
> Warning: Permanently added the RSA host key for IP address '140.82.114.3' to the list of known hosts.
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
> Submodule 'Cores/DSDeltaCore' (https://github.com/rileytestut/DSDeltaCore.git) registered for path 'Cores/DSDeltaCore'
> Submodule 'Cores/DeltaCore' (git@github.com:rileytestut/DeltaCore.git) registered for path 'Cores/DeltaCore'
> Submodule 'Cores/GBADeltaCore' (git@github.com:rileytestut/GBADeltaCore.git) registered for path 'Cores/GBADeltaCore'
> Submodule 'Cores/GBCDeltaCore' (git@github.com:rileytestut/GBCDeltaCore.git) registered for path 'Cores/GBCDeltaCore'
> Submodule 'Cores/N64DeltaCore' (git@github.com:rileytestut/N64DeltaCore.git) registered for path 'Cores/N64DeltaCore'
> Submodule 'Cores/NESDeltaCore' (git@github.com:rileytestut/NESDeltaCore.git) registered for path 'Cores/NESDeltaCore'
> Submodule 'Cores/SNESDeltaCore' (git@github.com:rileytestut/SNESDeltaCore.git) registered for path 'Cores/SNESDeltaCore'
> Submodule 'External/Harmony' (https://github.com/rileytestut/Harmony.git) registered for path 'External/Harmony'
> Submodule 'External/Roxas' (git@github.com:rileytestut/Roxas.git) registered for path 'External/Roxas'
> Submodule path 'Cores/DSDeltaCore': checked out '40d515a2088be952b6c4038ff715d1b91ce0667f'
> Submodule 'desmume' (https://github.com/TASVideos/desmume.git) registered for path 'desmume'
> Submodule path 'Cores/DSDeltaCore/desmume': checked out '9e6b4f128f45f84155c6e971e464de84c88986a7'
> Cloning into 'Cores/DeltaCore'...
> warning: templates not found /usr/local/git/share/git-core/templates
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
> Cloning into 'Cores/DeltaCore'...
> warning: templates not found /usr/local/git/share/git-core/templates
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
> Cloning into 'Cores/DeltaCore'...
> warning: templates not found /usr/local/git/share/git-core/templates
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:rileytestut/DeltaCore.git' into submodule path 'Cores/DeltaCore' failed
> Completed with errors, see above
> ```

This is unrelated. Please follow this: https://gist.github.com/adamjohnson/5682757

**Comment by valeIT** [comment]
Changing all ssh to https or cloning rcursively manually using a script works

**Comment by vhqtvn** [comment]
I got it to clone all by cd to each error submodule and do `git checkout master`, only for `External/Harmony` you'll need to run `git submodule update --init --recursive` again inside it.

After that when building you'll get undefined error for `ConflictResolution`, just change all of them to `SyncCoordinator.ConflictResolution` and the corresponding return enum value from `.newest` to `.local` and then it will compile.

Just a temporary workaround until @rileytestut push his updates.

**Comment by Ayukito** [comment]
This was fixed in the 1.1 push
